text_generation/causal_lm/cpp/continuous_batching/library/include/generation_handle.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+
+#include "generation_config.hpp"
+
+
+enum class GenerationStatus {
+    RUNNING = 0, // Default status for ongoing generation
+    FINISHED = 1, // Status set when generation has been finished
+    IGNORED = 2, // Status set when generation run into out-of-memory condition and could not be continued
+    DROPPED_BY_PIPELINE = 3, // Currently not used, TODO: implement abort functionality
+    DROPPED_BY_HANDLE = 4 // Status set when generation handle is dropped
```

Comment:
is there any difference between these 2 dropped statuses? I supposed abortion is the same as stream->drop() ?

Reply:
I added both, because both pipeline and handle can set and read that status and from the handle perspective it might be important to inform the client what happened. For now we could have just one status. If in the near future we don't implement the other status, I think we could get rid of it.